src/app.js

```diff
@@ -1,21 +1,25 @@
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
+const fs = require('fs');
+
+function mv() {
+  const [sourse, destination] = process.argv.slice(2);
+  const isDir = destination.slice(-1) === '/';
+  const finalDestination = isDir
+    ? destination + sourse.split('/').slice(-1)
+    : destination + '/' + sourse.split('/').slice(-1);
+
+  try {
+    fs.copyFileSync(sourse, finalDestination);
+    fs.rmSync(sourse);
+  } catch (error) {
+    if (error.code === 'ENOENT' && !isDir) {
+      fs.copyFileSync(sourse, destination);
+      fs.rmSync(sourse);
+    } else {
```

Comment:
you don't catch the error here. And can't you get the right destination on the first time?

Reply:
For the first time, I'm trying to move the source file to the destination supposing it could be an existing directory. If I fail, it could mean that the destination is a new file name. That's why I need to catch errors for the first time and implement some checks before trying to change the file name. If I fail again or my checks failed, it means that an error occurred and I throw it to show that something went wrong.